File: feature/appcenter/src/main/java/net/m3mobile/feature/appcenter/internal/AppCenterPasswordPolicy.kt
package net.m3mobile.feature.appcenter.internal

internal object AppCenterPasswordPolicy {

    fun assertCurrentPassword(currentPassword: String) {
        require(currentPassword.isNotEmpty()) {
            "Current AppCenter kiosk admin password is required."
        }
    }

    fun assertNewPassword(newPassword: String) {
        require(newPassword.length in MIN_NEW_PASSWORD_LENGTH..MAX_NEW_PASSWORD_LENGTH) {
            "New AppCenter kiosk admin password length must be 4 to 20 characters."
        }
    }

    private const val MIN_NEW_PASSWORD_LENGTH = 4
    private const val MAX_NEW_PASSWORD_LENGTH = 20
}

File: dotnet/M3Mobile.M3Sdk.Xamarin/AppCenter/AppCenterPasswordPolicy.cs
using System;

namespace M3Sdk.Xamarin.AppCenter
{
    internal static class AppCenterPasswordPolicy
    {
        private const int MinNewPasswordLength = 4;
        private const int MaxNewPasswordLength = 20;

        internal static void AssertCurrentPassword(string currentPassword)
        {
            if (string.IsNullOrEmpty(currentPassword))
                throw new ArgumentException("Current AppCenter kiosk admin password is required.", nameof(currentPassword));
        }

        internal static void AssertNewPassword(string newPassword)
        {
            if (newPassword == null ||
                newPassword.Length < MinNewPasswordLength ||
                newPassword.Length > MaxNewPasswordLength)
            {
                throw new ArgumentException(
                    "New AppCenter kiosk admin password length must be 4 to 20 characters.",
                    nameof(newPassword));
            }
        }
    }
}
